cpp/src/strings/regex/regcomp.cpp

```diff
@@ -484,86 +503,141 @@ class regex_parser {
+        auto bytes_read = transform_until(_expr_ptr, _expr_ptr + max_read, buffer.data(), "},");
+        if (_expr_ptr[bytes_read] != '}' && _expr_ptr[bytes_read] != ',') {
           break;  // re-interpret as CHAR
         }
         auto count = std::atoi(buffer.data());
-        CUDF_EXPECTS(count <= max_value,
-                     "unsupported repeat value at " + std::to_string(exprp - pattern - 1));
-        yy_min_count = static_cast<int16_t>(count);
+        CUDF_EXPECTS(
+          count <= max_value,
+          "unsupported repeat value at " + std::to_string(_expr_ptr - _pattern_begin - 1));
+        _min_count = static_cast<int16_t>(count);
 
-        auto const exprp_backup = exprp;  // save in case ending '}' is not found
-        exprp += bytes_read;
+        auto const expr_ptr_save = _expr_ptr;  // save in case ending '}' is not found
+        _expr_ptr += bytes_read;
 
         // get optional right-side (m) value => max_count
-        yy_max_count = yy_min_count;
-        if (*exprp++ == ',') {
-          bytes_read = transform_until(exprp, exprp + max_read, buffer.data(), "}");
-          if (exprp[bytes_read] != '}') {
-            exprp = exprp_backup;  // abort, rollback and
-            break;                 // re-interpret as CHAR
+        _max_count = _min_count;
+        if (*_expr_ptr++ == ',') {
+          bytes_read = transform_until(_expr_ptr, _expr_ptr + max_read, buffer.data(), "}");
+          if (_expr_ptr[bytes_read] != '}') {
+            _expr_ptr = expr_ptr_save;  // abort, rollback and
+            break;                      // re-interpret as CHAR
           }
 
           count = std::atoi(buffer.data());
-          CUDF_EXPECTS(count <= max_value,
-                       "unsupported repeat value at " + std::to_string(exprp - pattern - 1));
+          CUDF_EXPECTS(
+            count <= max_value,
+            "unsupported repeat value at " + std::to_string(_expr_ptr - _pattern_begin - 1));
 
           // {n,m} and {n,} are both valid
-          yy_max_count = buffer[0] == 0 ? -1 : static_cast<int16_t>(count);
-          exprp += bytes_read + 1;
+          _max_count = buffer[0] == 0 ? -1 : static_cast<int16_t>(count);
+          _expr_ptr += bytes_read + 1;
         }
 
         // {n,m}? pattern is lazy counted quantifier
-        if (*exprp == '?') {
-          exprp++;
+        if (*_expr_ptr == '?') {
+          _expr_ptr++;
           return COUNTED_LAZY;
         }
         // otherwise, fixed counted quantifier
         return COUNTED;
       }
       case '|': return OR;
     }
+    _chr = chr;
     return CHAR;
   }
 
- public:
-  /**
-   * @brief Single parsed pattern element.
-   */
-  struct Item {
-    int t;
-    union {
-      char32_t yy;
-      int yyclass_id;
-      struct {
-        short n;
-        short m;
-      } yycount;
-    } d;
-  };
-  std::vector<Item> m_items;
+  std::vector<regex_parser::Item> expand_counted_items()
```

Comment:
This is not a new function but was simply moved from the `regex_compiler` class to here since it only needs the `regex_parser::_items` variable.